pkg/controllers/node/health/controller.go

```diff
@@ -68,7 +72,35 @@ func NewController(kubeClient client.Client, cloudProvider cloudprovider.CloudPr
 func (c *Controller) Register(_ context.Context, m manager.Manager) error {
 	return controllerruntime.NewControllerManagedBy(m).
 		Named("node.health").
-		For(&corev1.Node{}, builder.WithPredicates(nodeutils.IsManagedPredicateFuncs(c.cloudProvider))).
+		For(&corev1.Node{}, builder.WithPredicates(nodeutils.IsManagedPredicateFuncs(c.cloudProvider), predicate.Funcs{
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				if len(e.ObjectOld.(*corev1.Node).Status.Conditions) != len(e.ObjectNew.(*corev1.Node).Status.Conditions) {
```

Comment:
nit: Can you just typecast things into variables like

```
oldNode := e.ObjectOld.(*corev1.Node)
newNode := e.ObjectNew.(*corev1.Node)
```

I think it will make things easier to read overall